src/App.jsx

```diff
@@ -1,44 +1,64 @@
-import { useState } from 'react';
-
+import { useState, useEffect } from 'react';
 import './App.scss';
-
-// Let's talk about using index.js and some other name in the component folder.
-// There's pros and cons for each way of doing this...
-// OFFICIALLY, we have chosen to use the Airbnb style guide naming convention. 
-// Why is this source of truth beneficial when spread across a global organization?
 import Header from './Components/Header';
 import Footer from './Components/Footer';
 import Form from './Components/Form';
 import Results from './Components/Results';
-
+import axios from 'axios';
 
 function App() {
   const [data, setData] = useState(null);
   const [requestParams, setRequestParams] = useState({});
   const [loading, setLoading] = useState(false);
 
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        if (requestParams.url && requestParams.method) {
+          setLoading(true);
+          let response;
+
+          switch (requestParams.method) {
+            case 'GET':
+              response = await axios.get(requestParams.url);
+              break;
+            case 'POST':
+              response = await axios.post(requestParams.url, requestParams.json);
+              break;
+            case 'PUT':
+              response = await axios.put(requestParams.url, requestParams.json);
+              break;
+            case 'DELETE':
+              response = await axios.delete(requestParams.url);
+              break;
+            default:
+              throw new Error('Invalid request method.');
+          }
```

Comment:
Since we've got our `requestParams` in a nice object package, we can collapse this big switch statement down to a single [axios command](https://axios-http.com/docs/api_intro) with a [config object](https://axios-http.com/docs/req_config) that conveniently matches our `requestParams`.